--- a/lib/Driver/ToolChains/CommonArgs.cpp	(e906cee58837a946d914c4ec49e7ba1901735ae4)
+++ b/lib/Driver/ToolChains/CommonArgs.cpp	(working tree)
@@ -858,7 +858,7 @@ collectSanitizerRuntimes(const ToolChain &TC, const ArgList &Args,
     StaticRuntimes.push_back("stats_client");
 
   // Always link the static runtime regardless of DSO or executable.
-  if (SanArgs.needsAsanRt())
+  if (SanArgs.needsAsanRt() && SanArgs.linkRuntimes())
     HelperStaticRuntimes.push_back("asan_static");
 
   // Collect static runtimes.
