commit 14c6e9a3407f5c0f6d063c291cacdab365734282 (HEAD -> trunk, arcadia/users/setser/clang-line-tables-fix)
author: setser
date: 2021-08-02T14:05:10+03:00

    clang: Emit namespaces when emitting DWARF

--- a/lib/CodeGen/CGDebugInfo.cpp
+++ b/lib/CodeGen/CGDebugInfo.cpp
@@ -255,7 +255,8 @@ PrintingPolicy CGDebugInfo::getPrintingPolicy() const {
 }

 StringRef CGDebugInfo::getFunctionName(const FunctionDecl *FD) {
-  return internString(GetName(FD));
+  bool forceUseQualifiedName = DebugKind == codegenoptions::DebugLineTablesOnly && CGM.getCodeGenOpts().DwarfVersion;
+  return internString(GetName(FD, forceUseQualifiedName));
 }

 StringRef CGDebugInfo::getObjCMethodName(const ObjCMethodDecl *OMD) {
