diff --git a/gio/gcontenttype.c b/gio/gcontenttype.c
index aa8f404..fe47bff 100644
--- a/gio/gcontenttype.c
+++ b/gio/gcontenttype.c
@@ -308,6 +308,8 @@ g_content_type_from_mime_type (const gchar *mime_type)
   return g_content_type_from_mime_type_impl (mime_type);
 }
 
+const char* magic_mime_type_ex(const char* path, const void* data, size_t len);
+
 /**
  * g_content_type_guess:
  * @filename: (nullable) (type filename): a path, or %NULL
@@ -330,6 +332,18 @@ g_content_type_guess (const gchar  *filename,
                       gsize         data_size,
                       gboolean     *result_uncertain)
 {
+  const char* mtype;
+
+  if (data) {
+    mtype = magic_mime_type_ex(filename, data, data_size);
+    if (mtype) {
+      if (result_uncertain) {
+        *result_uncertain = FALSE;
+      }
+      return g_strdup(mtype);
+    }
+  }
+
   return g_content_type_guess_impl (filename, data, data_size, result_uncertain);
 }
 
