diff --git a/gdk/wayland/gdkdisplay-wayland.c b/gdk/wayland/gdkdisplay-wayland.c
index 1b6f728..721afbf 100644
--- a/gdk/wayland/gdkdisplay-wayland.c
+++ b/gdk/wayland/gdkdisplay-wayland.c
@@ -1088,6 +1088,7 @@ _gdk_wayland_display_load_cursor_theme (GdkWaylandDisplay *display_wayland)
 {
   guint size;
   const char *name;
+  const char* env;
   GValue v = G_VALUE_INIT;
   gint64 before G_GNUC_UNUSED;
 
@@ -1099,8 +1100,15 @@ _gdk_wayland_display_load_cursor_theme (GdkWaylandDisplay *display_wayland)
   g_value_init (&v, G_TYPE_INT);
   if (gdk_display_get_setting (GDK_DISPLAY (display_wayland), "gtk-cursor-theme-size", &v))
     size = g_value_get_int (&v);
-  else
-    size = 32;
+  else {
+    env = getenv("XCURSOR_SIZE");
+
+    if (env) {
+      size = atoi(env);
+    } else {
+      size = 32;
+    }
+  }
   g_value_unset (&v);
 
   g_value_init (&v, G_TYPE_STRING);
