diff --git a/src/vulkan/wsi/wsi_common_wayland.c b/src/vulkan/wsi/wsi_common_wayland.c
index 928ec216505..19b4e0b34bd 100644
--- a/src/vulkan/wsi/wsi_common_wayland.c
+++ b/src/vulkan/wsi/wsi_common_wayland.c
@@ -1802,8 +1802,10 @@ wsi_wl_surface_create_swapchain(VkIcdSurfaceBase *icd_surface,
    wsi_wl_surface->chain = chain;
 
    result = wsi_wl_surface_init(wsi_wl_surface, wsi_device);
-   if (result != VK_SUCCESS)
+   if (result != VK_SUCCESS) {
+      fprintf(stderr, "FAILED TO wsi_wl_surface_init!\n");
       goto fail;
+   }
 
    enum wsi_wl_buffer_type buffer_type;
    struct wsi_base_image_params *image_params = NULL;
@@ -1888,8 +1890,10 @@ wsi_wl_surface_create_swapchain(VkIcdSurfaceBase *icd_surface,
    for (uint32_t i = 0; i < chain->base.image_count; i++) {
       result = wsi_wl_image_init(chain, &chain->images[i],
                                  pCreateInfo, pAllocator);
-      if (result != VK_SUCCESS)
+      if (result != VK_SUCCESS) {
+         fprintf(stderr, "FAILED TO wsi_wl_image_init!\n");
          goto fail_image_init;
+      }
       chain->images[i].busy = false;
    }
