diff --git a/glib/gquark.c b/glib/gquark.c
index 2799b7d..de66a64 100644
--- a/glib/gquark.c
+++ b/glib/gquark.c
@@ -174,7 +174,7 @@ quark_strdup (const gchar *string)
 
   return copy;
 }
-
+void glib_init(void);
 /* HOLDS: quark_global_lock */
 static inline GQuark
 quark_from_string (const gchar *string,
@@ -182,6 +182,8 @@ quark_from_string (const gchar *string,
 {
   GQuark quark = 0;
 
+  glib_init();
+
   quark = GPOINTER_TO_UINT (g_hash_table_lookup (quark_ht, string));
 
   if (!quark)
diff --git a/gobject/gtype.c b/gobject/gtype.c
index 0714299..18d09f6 100644
--- a/gobject/gtype.c
+++ b/gobject/gtype.c
@@ -4426,6 +4426,8 @@ g_type_init (void)
   g_assert_type_system_initialized ();
 }
 
+static int ginitdone;
+
 static void
 gobject_init (void)
 {
@@ -4434,6 +4436,12 @@ gobject_init (void)
   TypeNode *node;
   GType type G_GNUC_UNUSED  /* when compiling with G_DISABLE_ASSERT */;
 
+  if (ginitdone) {
+    return;
+  }
+
+  ginitdone = 1;
+
   /* Ensure GLib is initialized first, see
    * https://bugzilla.gnome.org/show_bug.cgi?id=756139
    */
@@ -4521,6 +4529,12 @@ gobject_init (void)
   _g_signal_init ();
 }
 
+void g_object_init(void);
+
+void g_object_init(void) {
+    gobject_init();
+}
+
 #ifdef G_PLATFORM_WIN32
 
 void gobject_win32_init (void);
