diff --git a/llvm/utils/TableGen/CodeGenSchedule.cpp b/llvm/utils/TableGen/CodeGenSchedule.cpp
index ee52b2e..eb654e3 100644
--- a/llvm/utils/TableGen/CodeGenSchedule.cpp
+++ b/llvm/utils/TableGen/CodeGenSchedule.cpp
@@ -370,7 +370,7 @@ processSTIPredicate(STIPredicateFunction &Fn,
                const std::pair<APInt, APInt> &LhsMasks = OpcodeMasks[LhsIdx];
                const std::pair<APInt, APInt> &RhsMasks = OpcodeMasks[RhsIdx];
 
-               auto LessThan = [](const APInt &Lhs, const APInt &Rhs) {
+               auto LessThan = [](const auto &Lhs, const auto &Rhs) {
                  unsigned LhsCountPopulation = Lhs.countPopulation();
                  unsigned RhsCountPopulation = Rhs.countPopulation();
                  return ((LhsCountPopulation < RhsCountPopulation) ||
