diff --git a/gdk/wayland/gdkdisplay-wayland.c b/gdk/wayland/gdkdisplay-wayland.c
index 20f662d..7d5995d 100644
--- a/gdk/wayland/gdkdisplay-wayland.c
+++ b/gdk/wayland/gdkdisplay-wayland.c
@@ -1165,6 +1165,7 @@ _gdk_wayland_display_load_cursor_theme (GdkWaylandDisplay *display_wayland)
 {
   guint size;
   const gchar *name;
+  const char* env;
   GValue v = G_VALUE_INIT;
 
   g_assert (display_wayland);
@@ -1173,8 +1174,15 @@ _gdk_wayland_display_load_cursor_theme (GdkWaylandDisplay *display_wayland)
   g_value_init (&v, G_TYPE_INT);
   if (gdk_screen_get_setting (display_wayland->screen, "gtk-cursor-theme-size", &v))
     size = g_value_get_int (&v);
-  else
-    size = 32;
+  else {
+    env = getenv("XCURSOR_SIZE");
+
+    if (env) {
+      size = atoi(env);
+    } else {
+      size = 32;
+    }
+  }
   g_value_unset (&v);
 
   g_value_init (&v, G_TYPE_STRING);
