--- qw
+++ dnssd_clientstub.c
@@ -2053,7 +2053,7 @@
     hdr = create_hdr(reg_service_request, &len, &ptr, (*sdRef)->primary ? 1 : 0, *sdRef);
     if (!hdr) { DNSServiceRefDeallocate(*sdRef); *sdRef = NULL; return kDNSServiceErr_NoMemory; }
     if (!callBack) hdr->ipc_flags |= IPC_FLAGS_NOREPLY;
-
+    const uint8_t* limit = ptr + len;
     put_flags(flags, &ptr);
     put_uint32(interfaceIndex, &ptr);
     put_string(name, &ptr);
@@ -2388,6 +2388,7 @@
     //If kDNSServiceFlagsQueueRequest flag is set, do not make separate return socket.
     hdr = create_hdr(reg_record_request, &len, &ptr, !(flags & kDNSServiceFlagsQueueRequest), sdRef);
     if (!hdr) return kDNSServiceErr_NoMemory;
+    const uint8_t* limit = ptr + len;
 
     put_flags(flags, &ptr);
     put_uint32(interfaceIndex, &ptr);
